src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/SampleSvcClient.java

```diff
@@ -1,24 +1,135 @@
 package uk.gov.ons.ctp.response.collection.exercise.client;
 
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
 import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.rest.RestUtility;
+import uk.gov.ons.ctp.response.collection.exercise.config.AppConfig;
 import uk.gov.ons.ctp.response.collection.exercise.domain.CollectionExercise;
+import uk.gov.ons.ctp.response.collection.exercise.domain.SampleLink;
+import uk.gov.ons.ctp.response.collection.exercise.repository.SampleLinkRepository;
+import uk.gov.ons.ctp.response.sample.representation.CollectionExerciseJobCreationRequestDTO;
 import uk.gov.ons.ctp.response.sample.representation.SampleSummaryDTO;
 import uk.gov.ons.ctp.response.sample.representation.SampleUnitsRequestDTO;
+import uk.gov.ons.response.survey.representation.SurveyDTO;
 
-/** Service responsible for making client calls to the Sample service */
-public interface SampleSvcClient {
+/** HTTP RestClient implementation for calls to the Sample service */
+@Component
+public class SampleSvcClient {
+  private static final Logger log = LoggerFactory.getLogger(SampleSvcClient.class);
+
+  @Autowired private AppConfig appConfig;
+
+  @Autowired private RestTemplate restTemplate;
+
+  @Autowired private SampleLinkRepository sampleLinkRepository;
+
+  @Qualifier("sampleRestUtility")
+  @Autowired
+  private RestUtility restUtility;
+
+  @Autowired private SurveySvcClient surveyService;
 
   /**
    * Request the delivery of sample units from the Sample Service via a message queue.
    *
    * @param exercise the Collection Exercise for which to request sample units.
    * @return the total number of sample units in the collection exercise.
    */
-  SampleUnitsRequestDTO requestSampleUnits(CollectionExercise exercise) throws CTPException;
+  @Retryable(
```

Comment:
The `@Retryable` annotation should be used very sparingly. We can't have users waiting 25 seconds for a response.